composer/datasets/ffcv_utils.py

```diff
@@ -80,7 +78,4 @@ def write_ffcv_dataset(dataset: Optional[Dataset] = None,
             ffcv.fields.IntField()
     },
                                        num_workers=num_workers)
-    if dataset:
-        writer.from_indexed_dataset(dataset, chunksize=chunk_size)
-    elif remote is not None:
-        raise NotImplementedError("FFCV with remote datasets is not implemented")
+    writer.from_indexed_dataset(dataset, chunksize=chunk_size)
```

Comment:
So.. `StreamingDataset` technically has a `__getitem__` function but it's not actually download-safe currently... it's sort of called internally only on "safe" downloaded indices. I see a couple options

* fix this @knighton 
* pre-download the `StreamingDataset`, I think we could spin a multi-worker DataLoader and it should only take a few min, and then you can run `writer.from_indexed_dataset` as-is. @dskhudia
* implement a `.from_iterable_datset` function to the FFCV writer.. but this seems like a pain

Reply:
Implemented #2 here.